### PR TITLE
 Linear regression module added 

### PR DESCRIPTION
**Pull request by Abhinaba006**
Linear regression module has added, it has fit and predict as sklearn module has and uses pandas and numpy for data manipulation and calculation.

### CODE REVIEW

File: Linear Regression/Linear_Regression.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import matplotlib.pyplot as plt
+
+class LinearRegression:
+    
+    def __init__(self):
+        self.costs = []
+        self.b = 0
+        self.w = []
+        
+    def forward(self, x, w, b):
```
Comment (ashwani-rathee) [inline Linear Regression/Linear_Regression.py:21]: Most of the issues(6 out of 10) come from this:https://docs.quantifiedcode.com/python-anti-patterns/correctness/method_could_be_a_function.html
Python raises the 'Method could be a function' error whenever you don't use self in functions inside class

File: Linear Regression/Linear_Regression.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
+"""
+Linear regression
+Arguments - alpha (by default .001)
+            epochs(by default 100)
+            plot : bool - plot the cost vs epoch graph
```
Comment (ashwani-rathee) [inline Linear Regression/Linear_Regression.py:5]: Trailing whitespace issue in line 5

File: Linear Regression/Linear_Regression.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+function -  fit : (x, y) - train the data
+            predict : (x) - predict the new y using previus trained model
+            this module also normalize data brfore training that increase training efficiency
+"""
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+
+class LinearRegression:
+
```
Comment (ashwani-rathee) [inline Linear Regression/Linear_Regression.py:15]: Trailing whitespace issue in line 15

File: Linear Regression/Linear_Regression.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            predict : (x) - predict the new y using previus trained model
+            this module also normalize data brfore training that increase training efficiency
+"""
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+
+class LinearRegression:
+    
+    def __init__(self):
```
Comment (ashwani-rathee) [inline Linear Regression/Linear_Regression.py:16]: You have docstring issue in here. Missing docstring in __init__

**Review by ashwani-rathee — COMMENTED** [review]
this should fixup most of your linting issues

**Comment by psyduck1203** [comment]
Hey @Abhinaba006. Sorry but I have worked on this issue from day 1 and pulled a request 7 days ago without conflicts #9. You can work on any other issue. I hope you will understand. Thank you!

**Comment by Abhinaba006** [comment]
> Hey @Abhinaba006. Sorry but I have worked on this issue from day 1 and pulled a request 7 days ago without conflicts #9. You can work on any other issue. I hope you will understand. Thank you!

So what can I do?

File: Linear Regression/Linear_Regression.py (hunk 4 above)
Comment (Abhinaba006) [reply Linear Regression/Linear_Regression.py:16]: Yes, I will fix that

**Comment by psyduck1203** [comment]
You can work on any other issue

**Comment by ashwani-rathee** [comment]
@psyduck1203 I am sorry for the confusion,but were you properly assigned to the base setup by any of the mentors or project admin?

**Comment by ashwani-rathee** [comment]
@geekquad What should we do in this case?

**Comment by psyduck1203** [comment]
@ashwani-rathee I am active from day 1 in this Repo you can check the date and time of the pr made by me also you can check the closed issues where I asked the maintainers to assign me the issue but no one replied. Then I tried to contact @geekquad and @plazzy99 on discord as only both of them were named as the maintainers in the readme and till the time they reply all this happened.

File: Linear Regression/Linear_Regression.py (hunk 1 above)
Comment (Abhinaba006) [reply Linear Regression/Linear_Regression.py:21]: so can I remove self in the function?

**Comment by Abhinaba006** [comment]
@ashwani-rathee sir, did any more changes needed? I have fixed the problems that you told me.

**Comment by ashwani-rathee** [comment]
@Abhinaba006 relax buddy,I am supposed to be more careful with base implementation,that's why it's taking a little more time :)

**Comment by Abhinaba006** [comment]
Okk

**Comment by ashwani-rathee** [comment]
This looks good to me.@geekquad what do you think?

**Review by geekquad — APPROVED** [review]
Amazing work @Abhinaba006.

**Pull request merged**